mesonbuild/coredata.py

```diff
@@ -1069,6 +1069,33 @@ def parse_cmd_line_options(args: argparse.Namespace) -> None:
             args.cmd_line_options[key] = value
             delattr(args, name)
 
+def validate_dirs(builddir: T.Optional[str], builddir_or_sourcedir: T.Optional[str]) -> T.Tuple[T.Optional[Path], T.Optional[Path]]:
+    if builddir is not None:
+        if builddir_or_sourcedir is not None:
+            raise MesonException('builddir defined with -C argument and positional argument')
```

Comment:
Perhaps "directory specified with both -C and a positional argument".

Reply:
ok, updated.